### PR TITLE
Fabricate extended CMIP7 historical coverage for test data

### PR DESCRIPTION
**Pull request by lewisjared**
CMIP7 test data is converted from CMIP6, so fabricated CMIP7 `historical` runs only reach ~2014. Diagnostics whose CMIP7 window extends past 2014 — notably climate-drivers-for-fire (2002–2021) — could therefore neither be solved in the unit tests nor have their CMIP7 baseline minted.

This adds an opt-in mechanism to fabricate extended CMIP7 historical coverage, without touching any diagnostic whose window ends in or before 2014.

**Mint / conversion path.** `shift_time_axis_end` relabels a monthly cftime axis by a constant whole-month offset, preserving calendar and spacing. It is plumbed through `CMIP7Request(extend_historical_to=(year, month))` and the CMIP6→CMIP7 file conversion, defaulting off so every existing CMIP7 conversion is byte-identical.

**Solve-test catalog.** `scripts/generate_cmip7_catalog.py` gains an option to lift full-length historical rows (those ending exactly 2014-12) to a later end date, leaving shorter runs and all other datasets untouched. `tests/test-data/esgf-catalog/cmip7_catalog.parquet` is regenerated with historical coverage extended to 2021.

Because every other esmvaltool/pmp/ilamb diagnostic uses a window ending ≤2014, extending coverage past 2014 changes nothing for them: the full solve-regression suite passes with zero snapshot changes. New unit tests cover the shift logic.

This unblocks the CMIP7 baseline for the climate-drivers-for-fire diagnostic (see #670), which depends on this change.

## Summary by CodeRabbit

* **New Features**
  * Added an opt-in option to extend CMIP7 `historical` coverage beyond 2014 (up to 2021), including updated request and catalog-generation flows.
  * Added support for relabelling the CMIP time axis to a requested end year/month while preserving monthly spacing.
* **Bug Fixes**
  * ENSO CMIP7 recipe timeranges were updated to reflect the longer historical period.
  * Time coordinate updates keep associated time bounds consistent.
* **Tests**
  * Added unit tests covering the time-axis relabelling behaviour and default/no-op behaviour.

### CODE REVIEW

File: packages/climate-ref-core/src/climate_ref_core/cmip6_to_cmip7.py [hunk 1]
```diff
@@ -483,9 +485,90 @@ def convert_cmip6_to_cmip7_attrs(
...
+    calendar = last.calendar  # type: ignore[attr-defined]
+
+    def _shift(t: cftime.datetime) -> cftime.datetime:
+        total = _month_index(t) + offset_months
+        year, month = divmod(total, 12)
+        # Rebuild via cftime with the original calendar, keeping day and sub-day
+        # fields, so the shifted axis matches the shape of a real monthly axis.
+        return cftime.datetime(  # type: ignore[call-arg]
+            year, month + 1, t.day, t.hour, t.minute, t.second, t.microsecond, calendar=calendar
+        )
```
[the remark this answers is not in the document]
Comment (lewisjared) [reply packages/climate-ref-core/src/climate_ref_core/cmip6_to_cmip7.py:567]: Fixed in d977e3d0: `shift_time_axis_end` now rejects `end_month` outside 1..12 and clamps the day-of-month to the target month's length (computed via cftime arithmetic, so it respects noleap/360_day/etc.), so a day-31 or leap-Feb-29 label shifted onto a shorter month no longer raises and silently skips the file. Added edge-case tests: invalid `end_month`, a non-December target, and the day-clamp case.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
